lib/kb_Metrics/metricsdb_controller.py

```diff
@@ -515,44 +512,43 @@ def join_app_task_ujs(self, exec_tasks, exec_apps, ujs_jobs):
             u_j_s['creation_time'] = u_j_s.pop('created')
             u_j_s['modification_time'] = u_j_s.pop('updated')
             u_j_s['estcompl'] = j.get('estcompl', None)
-            u_j_s['time_info'] = [u_j_s.get('creation_time'),
-                                  u_j_s.get('modification_time'),
-                                  u_j_s['estcompl']]
 
-            if u_j_s.get('authstrat', None) == 'kbaseworkspace':
-                u_j_s['wsid'] = u_j_s.get('authparam')
+            authparam = u_j_s.pop('authparam')
+            authstrat = u_j_s.pop('authstrat')
+            if authstrat == 'kbaseworkspace':
+                u_j_s['wsid'] = authparam
 
             if u_j_s.get('desc'):
-                desc = u_j_s['desc'].split()[-1]
+                desc = u_j_s.pop('desc').split()[-1]
                 if '.' in desc:
                     u_j_s['method'] = desc
 
             # Assuming complete, error and status all exist in the records returned
-            if not j.get('error'):
-                if j.get('complete'):
+            if not u_j_s.get('error'):
+                if u_j_s.get('complete'):
                     u_j_s['job_state'] = 'completed'
-                elif j.get('status') in ["Initializing", 'queued']:
-                    u_j_s['job_state'] = j['status']
-                elif j.get('status') in ['canceled', 'cancelled']:
+                elif u_j_s.get('status') in ["Initializing", 'queued']:
+                    u_j_s['job_state'] = u_j_s['status']
+                elif u_j_s.get('status') in ['canceled', 'cancelled']:
                     u_j_s['job_state'] = 'canceled'
-                elif j.get('started'):
+                elif u_j_s.get('exec_start_time'):
                     u_j_s['job_state'] = 'in-progress'
-                elif j['created'] == j['updated']:
+                elif u_j_s['creation_time'] == u_j_s['modification_time']:
                     u_j_s['job_state'] = 'not-started'
-                elif j['created'] < j['updated'] and not j.get('started'):
+                elif (u_j_s['creation_time'] < u_j_s['modification_time'] and
+                        not u_j_s.get('exec_start_time')):
                     u_j_s['job_state'] = 'queued'
                 else:
                     u_j_s['job_state'] = 'unknown'
             else:
-                u_j_s['job_state'] = 'suspended'
+                u_j_s['job_state'] = 'suspend'
 
             for lat in app_task_list:
-                if ObjectId(lat['ujs_job_id']) == j['_id']:
+                if ObjectId(lat['ujs_job_id']) == u_j_s['job_id']:
                     if 'job_state' not in u_j_s:
```

Comment:
this will never happen.

Reply:
Why?

Reply:
your previous block guarantee u_j_s['job_state'] is assigned something.

Reply:
Now that if we don't need to serve 'job_state' for the downstream, it does not matter any more.  This will be removed together with the previous block.  For the sake of explanation: when I first started to work on this project, I was pointed to the existing API's that served the wanted data.  The master one is Catalog, which has the APIs to supply data from exec_engine collections.  Until we found that only had successfully 'executed' jobs, but we want the ones that failed, queued, ... as well.  So, I am not purposefully adding them here to make it look grand :)